feedly/api_client/data.py

```diff
@@ -162,13 +175,62 @@ class Entry(FeedlyData):
     pass
 
 
+StreamableT = TypeVar("StreamableT", bound=Streamable)
+
+UUID_REGEX = re.compile(r"[0-9a-f]{8}\b-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-\b[0-9a-f]{12}", re.IGNORECASE)
+
+
+class LazyStreams(Generic[StreamableT]):
+    def __init__(
+        self, parts: List[str], endpoint: str, factory: Callable[[Dict, APIClient], StreamableT], client: APIClient
+    ):
+        self.parts = parts
+        self.client = client
+        self.endpoint = endpoint
+        self.factory = factory
+
+        self.populated = False
+
+    @cached_property
+    def streams(self) -> List[StreamableT]:
+        self.populated = True
+        return [self.factory(item, self.client) for item in self.client.do_api_request(self.endpoint)]
+
+    @cached_property
+    def id2stream(self) -> Dict[str, StreamableT]:
+        return {stream.stream_id.content_id: stream for stream in self.streams}
+
+    @cached_property
+    def name2stream(self) -> Dict[str, StreamableT]:
+        return {stream["label"]: stream for stream in self.streams}
+
+    def get(self, name_or_id: Union[str, StreamIdBase]) -> StreamableT:
+        if isinstance(name_or_id, StreamIdBase):
+            return self.make_stream_from_id(name_or_id.content_id)
```

Comment:
is this safe? should we check that the other parts are matching our expectation?